lib/linter/linter.js

```diff
@@ -837,16 +945,16 @@ const BASE_TRAVERSAL_CONTEXT = Object.freeze(
  * @param {SourceCode} sourceCode A SourceCode object for the given text
  * @param {Object} configuredRules The rules configuration
  * @param {function(string): Rule} ruleMapper A mapper function from rule names to rules
- * @param {Object} parserOptions The options that were passed to the parser
  * @param {string} parserName The name of the parser in the config
```

Comment:
I want to break them. :) Passing an empty string or other string value will be a more difficult error to catch that having it be undefined. Because this is only in the flat config path, it will help us identify rules that rely on this early.